File: sdks/dotnet/sulfone-helium/Api/Core/CyanReq.cs
using System.Dynamic;

namespace sulfone_helium.Api.Core;

public readonly struct CyanGlobReq
{
    public string? Root { get; init; }
    public string Glob { get; init; }
    public string[] Exclude { get; init; }
    public string Type { get; init; }
}

public readonly struct CyanPluginReq
{
    public string Name { get; init; }
    public ExpandoObject Config { get; init; }
}

public readonly struct CyanProcessorReq
{
    public string Name { get; init; }
    public ExpandoObject Config { get; init; }

    public CyanGlobReq[] Files { get; init; }
}

public readonly struct CyanReq
{
    public CyanProcessorReq[] Processors { get; init; }
    public CyanPluginReq[] Plugins { get; init; }
}

File: sdks/node/src/api/core/cyan_req.ts
interface CyanGlobReq {
  readonly root?: string;
  readonly glob: string;
  readonly exclude: string[];
  readonly type: string;
}

interface CyanPluginReq {
  readonly name: string;
  readonly config: unknown;
}

interface CyanProcessorReq {
  readonly name: string;
  readonly config: unknown; // This is similar to ExpandoObject
  readonly files: CyanGlobReq[];
}

interface CyanReq {
  readonly processors: CyanProcessorReq[];
  readonly plugins: CyanPluginReq[];
}

// export all
export type { CyanReq, CyanProcessorReq, CyanPluginReq, CyanGlobReq };
